base/process.jl

```diff
@@ -785,7 +796,7 @@ function process_status(s::Process)
     process_signaled(s) ? "ProcessSignaled("*string(s.termsignal)*")" :
     #process_stopped(s) ? "ProcessStopped("*string(process_stop_signal(s))*")" :
     process_exited(s) ? "ProcessExited("*string(s.exitcode)*")" :
-    error("process status error")
+    @assert(false, "process status error")
```

Comment:
This is kind of weird. `error` seems like a better choice here.

Reply:
assert is right if this is supposed to be a complete list of all statuss a process can have.
If not then a `UnknownProcessStatusError` or maybe a `DomainError` is suited